docs/sources/reference/run.md

```diff
@@ -507,6 +507,29 @@ We have four ways to set memory usage:
   </tbody>
 </table>
 
+Examples:
+
+    # docker run -ti ubuntu:14.04 /bin/bash
+
+We set nothing about memory, this means we can use as much memory and swap memory
+as we need.
+
+    # docker run -ti -m 300M --memory-swap -1 ubuntu:14.04 /bin/bash
+
+We set memory limit and disabled swap memory limit, this means we can use 300M
+memory and as much swap memory as we need (if the host supports swap memory).
+
+    # docker run -ti -m 300M ubuntu:14.04 /bin/bash
+
+We set memory limit only, this means we can use 300M memory and 300M swap memory,
+by default, memory-swap will be set as double of memory, in this case, memory + swap
+would be 2*300M, so we can use 300M swap memory as well.
```

Comment:
mmm, something needs tweaking here - I think perhaps changing the `memory-swap will be set to double of memory` makes it less clear that we're trying to say that the `--memory-swap` flag is the total virtual memory size, including the `--memory` portion.